jetty-websocket/websocket-javax-server/src/main/java/org/eclipse/jetty/websocket/javax/server/internal/JavaxWebSocketServerContainer.java

```diff
@@ -260,4 +262,19 @@ protected void doStart() throws Exception
             deferredEndpointConfigs.clear();
         }
     }
+
+    @Override
+    public CompletableFuture<Void> shutdown()
+    {
+        LifeCycle.stop(sessionTracker);
```

Comment:
Not sure how we should stop accepting new incoming connections before this.

Reply:
Can stopping the sessionTracker take time? Can it block? Can it call user code?   If any of the above, then this is precisely what we don't want to do in a shutdown thread.   Why can't this be done in doStop?  at the very least a comment is needed, but I think a different approach is required.

Reply:
It won't block in jetty code but it will the user code with `onClosed`. We already stop the `SessionTracker` when the server is stopped (in 9.4 as well) so if this is an issue it already exists in the codebase before this change.

The problem here is that `Server.doStop()` shuts down all the connectors before calling `super.doStop()` and this closes the existing connections. So we can't react to the stop and close the WebSocket sessions as all the existing connections have already been closed.